### PR TITLE
[REVIEW] Enable typecasting between decimal and int

### LINKED ISSUE #7440: [FEA] Support casting to DecimalDtypes in cuDF
**Is your feature request related to a problem? Please describe.**
Now that cuDF supports `decimal` columns it would be great to be able to cast to `DecimalDtype` from `float` or `int` datatypes.

For example:
```
import cudf
from cudf.core.dtypes import Decimal64Dtype

df = cudf.DataFrame({"a":[0.51,1.51,2.51]}, dtype="float64")
df.astype(Decimal64Dtype(7,2))
```

This currently throws:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/dataframe.py", line 1163, in astype
    dtype=dtype, **kwargs
  File "/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/column/column.py", line 1018, in astype
    elif np.issubdtype(dtype, np.datetime64):
  File "/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/numpy/core/numerictypes.py", line 388, in issubdtype
    arg1 = dtype(arg1).type
TypeError: Cannot interpret 'Decimal64Dtype(precision=7, scale=2)' as a data type

```

**Describe the solution you'd like**
I would like to be able to convert `float` or `int` columns to `decimal` columns with the specified scale and precision

**Describe alternatives you've considered**
I've also found that:

```
 df = cudf.DataFrame({"a":[0.51,1.51,2.51]}, dtype=Decimal64Dtype(7,2))
```
throws the same error

### PR DESCRIPTION
**Pull request by ChrisJar**
This enables type conversions between `decimal` types and `int` types.

Closes #7440

### CODE REVIEW

**Comment by ttnghia** [comment]
This is a workaround, not a fix. I would recommend figuring out what's wrong with the `libcudf.unary.cast` implementation instead.

**Comment by galipremsagar** [comment]
@ChrisJar Now that https://github.com/rapidsai/cudf/pull/7733 is merged, we can enable type-casting decimal types to and from integer types as part of this PR itself.

File: python/cudf/cudf/tests/test_decimal.py [hunk 1]
```diff
@@ -151,14 +194,15 @@ def test_typecast_to_from_decimal(data, from_dtype, to_dtype):
...
+    got = data.astype(from_dtype)
+    pa_arr = got.to_arrow().cast(to_dtype, safe=False)
 
-    actual = actual.astype(to_dtype)
+    got = got.astype(to_dtype)
     expected = cudf.Series(NumericalColumn.from_arrow(pa_arr))
 
-    assert_eq(actual, expected)
+    assert_eq(got, expected)
+    assert_eq(got.dtype, expected.dtype)
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_decimal.py:208]:
```suggestion
    assert_eq(got, expected, check_dtype=True)
```

You can validate the dtype & series in a single go like this. Similarly in other tests too..

Comment (ChrisJar) [reply python/cudf/cudf/tests/test_decimal.py:208]: So this issue is that even with `check_dtype=True` the `assert_eq` function doesn't check if precision and scale are equal, but `assert_eq(got.dtype, expected.dtype)` does.

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_decimal.py:208]: Ah, that is because doesn't have a decimal type yet..Then we should keep this for now.

**Review by galipremsagar — APPROVED** [review]

File: python/cudf/cudf/core/column/decimal.py [hunk 2]
```diff
@@ -80,19 +79,11 @@ def as_decimal_column(
     ) -> "cudf.core.column.DecimalColumn":
         if dtype == self.dtype:
             return self
-        result = libcudf.unary.cast(self, dtype)
-        if isinstance(dtype, cudf.core.dtypes.Decimal64Dtype):
-            result.dtype.precision = dtype.precision
-        return result
+        return libcudf.unary.cast(self, dtype)
```
Comment (isVoid) [inline python/cudf/cudf/core/column/decimal.py:82]: Just a quick side note that this is needed in https://github.com/rapidsai/cudf/pull/7716

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**